source/gui/speechDict.py

```diff
@@ -94,20 +96,37 @@ def onOk(self, evt):
 			)
 			self.patternTextCtrl.SetFocus()
 			return
+		entryType = self.getType()
 		try:
 			dictEntry = self.dictEntry = speechDictHandler.SpeechDictEntry(
 				self.patternTextCtrl.GetValue(),
 				self.replacementTextCtrl.GetValue(),
 				self.commentTextCtrl.GetValue(),
 				bool(self.caseSensitiveCheckBox.GetValue()),
-				self.getType()
+				entryType,
+			)
+		except RegexpError as e:
+			log.debugWarning(f"Could not add dictionary entry due to regex error in the pattern field : {e}")
+			if entryType != speechDictHandler.ENTRY_TYPE_REGEXP:
+				raise(e)
+			gui.messageBox(
+				# Translators: This is an error message to let the user know that the dictionary entry is not valid.
+				_(f"Regular Expression error in the pattern field: \"{e}\"."),
+				# Translators: The title of an error message raised by the Dictionary Entry dialog
+				_("Dictionary Entry Error"),
+				wx.OK | wx.ICON_WARNING,
+				self
 			)
+			return
+		try:
 			dictEntry.sub("test")  # Ensure there are no grouping error (#11407)
-		except Exception as e:
-			log.debugWarning("Could not add dictionary entry due to (regex error) : %s" % e)
+		except RegexpError as e:
+			log.debugWarning(f"Could not add dictionary entry due to regex error in the replacement field : {e}")
+			if entryType != speechDictHandler.ENTRY_TYPE_REGEXP:
+				raise(e)
 			gui.messageBox(
 				# Translators: This is an error message to let the user know that the dictionary entry is not valid.
-				_("Regular Expression error: \"%s\".") % e,
+				_("Regular Expression error in the replacement field: \"{e}\"."),
```

Comment:
Missing f in fstring

```suggestion
				_(f"Regular Expression error in the replacement field: \"{e}\"."),
```